### PR TITLE
correction filing (part 2)

### PR DESCRIPTION
**Pull request by severinbeauvais**
*Issue #:* /bcgov/entity#2654

*Description of changes:*
This is Part 2 of the the Correction Filing task, which includes:
- fixed a bug in Filing History List
- updated payment option behaviour per Scott
- updated Staff Payment in AR, COD and COA filings
- always show Staff Payment section for staff
- save/fetch new flags
- updated filing data logic to set flags per code

Part 3 will implement:
- resume a draft Correction filing
- updates to unit tests

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the business-filings-ui license (Apache 2.0).

### CODE REVIEW

File: src/views/AnnualReport.vue [hunk 1]
```diff
@@ -911,8 +963,7 @@ export default {
     },
 
     hasAction (director, action) {
-      if (director.actions.indexOf(action) >= 0) return true
-      else return false
+      return (director.actions.indexOf(action) >= 0)
```
Comment (cameron-freshworks) [inline src/views/AnnualReport.vue]: Nice clean up

File: src/views/StandaloneDirectorsFiling.vue [hunk 2]
```diff
@@ -632,20 +645,42 @@ export default {
...
+    /**
+     * Adds/removes codes or sets flags in the Filing Data object.
+     * @param addRemove Whether to add or remove the specified codes/flags.
+     * @param filingCode The Filing Type Code to add or remove (optional).
+     * @param priority The Priority flag to set or clear (optional).
+     * @param waiveFees The Waive Fees flag to set or clear (optional).
+     * @example setFilingData('add', undefined, undefined, true) -> adds Waive Fees to all codes
+     * @example setFilingData('remove', undefined, true, undefined) - removes Priority from all codes
+     * @example setFilingData('add', 'OTFDR', false, false) -> adds OTFDR code with neither flag
+     * @example setFilingData('add', 'OTCDR') -> adds OTCDR code with default flags
```
Comment (cameron-freshworks) [inline src/views/StandaloneDirectorsFiling.vue]: Extensive examples!

Comment (severinbeauvais) [reply src/views/StandaloneDirectorsFiling.vue]: The functionality is the same in all components but I tailored the examples :)
This is a candidate for a mixin, for sure. Maybe in my next commit :)

Comment (cameron-freshworks) [reply src/views/StandaloneDirectorsFiling.vue]: Sounds good to me!

**Pull request merged**